core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -490,7 +496,7 @@ class ByteBufferMessageSet(val buffer: ByteBuffer) extends MessageSet with Loggi
           Utils.writeUnsignedInt(buffer, MessageSet.LogOverhead + Message.CrcOffset, wrapperMessage.computeChecksum)
         }
         buffer.rewind()
-        (this, false)
+        ValidationAndOffsetAssignResult(this, maxTimestamp, false)
```

Comment:
It seems that maxTimestamp won't be set properly if messageTimestampType is LOG_APPEND_TIME?